### ISSUE TITLE
End effector transformations not coherent

### ISSUE THREAD

**Issue by fbottarel** [robotology/grasping-benchmarks-panda#2]
While using the code on the actual robot, I figured that the end effector poses commanded to the robot are somewhat imprecise along the approach axis. I think I might have found the cause: some grasp planners modify the grasping pose according to some hardcoded value along the approach axis:

- [the dexnet grasp planner](https://github.com/robotology/grasping-benchmarks-panda/blob/master/grasping_benchmarks/dexnet/dexnet_grasp_planner.py#L260) assumes an offset of 0.13 m
- [the superquadric planner](https://github.com/robotology/grasping-benchmarks-panda/blob/master/grasping_benchmarks/superquadric_based/superquadrics_grasp_planner.py#L342) assumes an offset of 0.1 m
- the gpd planner does not apply any offset.

We should normalize this behaviour. This code was originally written with the intention of using the `panda_hand` reference frame as an end effector, and according to the URDF files this frame is on the robot hand flange, not the actual gripper center. I modified the URDF and moveit configurations with a virtual link defining a tool center point (TCP for short) that will be used as a reference for grasping motions. 

tl;dr: I will modify the offsets hardcoded in this repo to point to the actual grasping point, not the wrist reference frame, so that it makes more sense with the new grasp server I'm writing.

**Comment by fbottarel** [comment]
I attempted to fix this in [this branch](https://github.com/robotology/grasping-benchmarks-panda/tree/explicit_grasp_offset). Everything works the same way, but now every planner has a variable that encodes a xyz translation in the EEF axis. 

The idea is to tune every grasp planner so that the planned grasp refers to the center of the tcp link and not the `panda_hand` link. I will test the code and then merge into master.
